cmd/okctl/hooks/state.go

```diff
@@ -230,6 +230,29 @@ func PurgeRemoteState(o *okctl.Okctl) RunEer {
 	}
 }
 
+// VerifyClusterExistsInState ensures we have a cluster in the state before continuing
+// This could happen if maintenance mode was not run after upgrading
+// to 0.0.80 and we don't have a state.db in S3 - see #491
+func VerifyClusterExistsInState(o *okctl.Okctl) RunEer {
+	return func(_ *cobra.Command, _ []string) error {
+		handlers := o.StateHandlers(o.StateNodes())
+
+		_, err := handlers.Cluster.GetCluster(o.Declaration.Metadata.Name)
+		if err != nil {
+			log := logging.GetLogger(logComponent, "VerifyClusterExistsInState")
+			log.Debug("Could not find cluster in current state")
+
+			if errors.Is(err, storm.ErrNotFound) {
+				return fmt.Errorf("getting existing cluster %s: %w", o.Declaration.Metadata.Name, err)
+			}
```

Comment:
From an UX perspective, I think we could give the user more assistance if this happens.

The change we did earlier was a breaking change, and we need the user to do execute some actions in order to fix the cluster. These are the maintenance commands, and we have described them in https://www.okctl.io/concerning-upgrades-and-patches/ .

So I think we could link to that article. For instance:

```golang
if errors.Is(err, storm.ErrNotFound) {
  _, _ = fmt.Fprintln(
    o.Out,
    "The state file for the cluster was not found. If you have recently upgraded from okctl version 0.0.80, "+
      "you need to do some manual maintenance before continuing. See: "+
      "https://www.okctl.io/concerning-upgrades-and-patches/",
  )
  return fmt.Errorf("getting existing cluster %s: %w", o.Declaration.Metadata.Name, err)
}
```

For the longer term, we have a plan to improve UX on most of error messages. Right now they suffer from being purely technical, and offer no solution. In some future, we could write a really user friendly error message, one technical, and offer a solution.

Reply:
What I did was create a commin issue section with the error message, so that you can search for `getting existing cluster` and get : https://www.okctl.io/common-issues/#i-get-an-error-message-with-getting-existing-cluster-cluster-name-not-found

Reply:
We could. Though I argumented against it due to:
1. We have no structure for delivering user oriented error messages, its a TODO
2. We by principle try to keep the code base focused on the current version, not looking backwards. 

Thats why I thought making it easy to search for the error on [okctl.io](https://okctl.io) was a nice compromise